include/DataflowPass.h

```diff
@@ -106,7 +107,72 @@ class DataflowPass {
 
     here, annoFnName = "free" and invokerFnName = "free0"
     */
-    virtual void onAnnotation(PVAliasSet* input, std::string &annoFnName, std::string& invokerFnName,
+    virtual void onAnnotation(PVAliasSet* input, const std::string &annoFnName, const std::string& invokerFnName,
+                              AnnotationType annotationType) = 0;
+
+    // annofnName is the name of the function inside the parameters.
+    // this overload exists (where we do not specify the function that invoked
+    // the call) because there are cases where it is not appropriate to add
+    // the invoker function to the called methods set
+    /*
+    Example: see the following code
+
+    -----------
+    struct my_struct {
+        char* x MustCall("free");
+        char* y MustCall("free");
+    };
+
+    my_struct Calls("free", "x") free_and_obligate(my_struct S Calls("free", "x")) {
+      free(S.x);
+      return S;
+    }
+
+    int main() {
+      my_struct A;
+      A.x = (char*)malloc(15);
+
+      my_struct B = free_and_obligate(A);
+    }
+    -----------
+
+    This code finds that there is a ReturnAnnotation on the function
+    free_and_obligate and will add "free" to return.x's called methods. Here, B is
+    set to that return value so B.x will have called methods {"free"}. But if we
+    also say to add the invoker name to the called methods (free_and_obligate),
+    then B.x's called methods will be {"free", "free_and_obligate"}. This is not
+    correct, hence the need to be able to include or and not include
+    the invoker name.
```

Comment:
I would expect that return annotations and the standard rules for adding dataflow facts do not interact in cases like this. In particular, I'd expect that the code handles the following entirely separately:
* process the fact that `free_and_obligate` is called on `A`. If we happen to also have some other fact that allows us to conclude that B and A are aliases, this fact can also be propagated to B. (In the Java RLC, this would be handled by the Returns Receiver checker, which would provide this kind of fact.)
* process the return annotation, which only applies to the `.x` field of the returned value; this should be propagated to `B.x` (and then `A.x` from there, again, iff A and B are known to be aliases).

The resulting dataflow store after this call thus looks like (assuming A and B are in fact aliases):
```
A = B -> CalledMethods("free_and_obligate")
A.x = B.x -> CalledMethods("free")
```

If A and B were not (known to be) aliases, we'd instead have:
```
A -> CalledMethods("free_and_obligate")
B -> no information/top
A.x -> no information/top
B.x -> CalledMethods("free")
```